src/lerobot/policies/pi0/modeling_pi0.py

```diff
@@ -257,6 +258,98 @@ def reset(self):
         """This should be called whenever the environment is reset."""
         self._action_queue = deque([], maxlen=self.config.n_action_steps)
 
+    @classmethod
+    def _transform_state_dict_keys(cls, state_dict: dict) -> dict:
+        """Transform state dict keys to match expected model structure."""
+        import re
+
+        transformed_dict = {}
+
+        for key, value in state_dict.items():
+            new_key = key
+            # Apply transformations for PaliGemma components
+            # model.paligemma_with_expert.paligemma.language_model.lm_head -> model.paligemma_with_expert.paligemma.lm_head
+            # model.paligemma_with_expert.paligemma.language_model.model -> model.paligemma_with_expert.paligemma.model.language_model
+            # model.paligemma_with_expert.paligemma.vision_tower -> model.paligemma_with_expert.paligemma.model.vision_tower
+            # model.paligemma_with_expert.paligemma.multi_modal_projector -> model.paligemma_with_expert.paligemma.model.multi_modal_projector
+            transformations = [
+                (
+                    re.compile(r"\.paligemma_with_expert\.paligemma\.language_model\.lm_head"),
+                    ".paligemma_with_expert.paligemma.lm_head",
+                ),
+                (
+                    re.compile(r"\.paligemma_with_expert\.paligemma\.language_model\.model"),
+                    ".paligemma_with_expert.paligemma.model.language_model",
+                ),
+                (
+                    re.compile(r"\.paligemma_with_expert\.paligemma\.vision_tower"),
+                    ".paligemma_with_expert.paligemma.model.vision_tower",
+                ),
+                (
+                    re.compile(r"\.paligemma_with_expert\.paligemma\.multi_modal_projector"),
+                    ".paligemma_with_expert.paligemma.model.multi_modal_projector",
+                ),
+            ]
+
+            for pattern, replacement in transformations:
+                new_key = pattern.sub(replacement, new_key)
+
+            transformed_dict[new_key] = value
+
+        # Handle tied weights: lm_head.weight and embed_tokens.weight share memory
+        # If we have lm_head.weight, also create embed_tokens.weight entry
+        lm_head_key = None
+        embed_tokens_key = None
+
+        for key in transformed_dict:
+            if key.endswith(".paligemma_with_expert.paligemma.lm_head.weight"):
+                lm_head_key = key
+            elif key.endswith(".paligemma_with_expert.paligemma.model.language_model.embed_tokens.weight"):
+                embed_tokens_key = key
```

Comment:
```suggestion
        for key in transformed_dict:
            if key.endswith(".paligemma_with_expert.paligemma.lm_head.weight"):
                lm_head_key = key
            elif key.endswith(".paligemma_with_expert.paligemma.model.language_model.embed_tokens.weight"):
                embed_tokens_key = key
            
            if lm_head_key and embed_tokens_key:
                break
```
Can we add early termination beacause it makes the code easier to read

Reply:
Thanks! I learned a lot.